Trabalho2/grafo.c

```diff
@@ -10,57 +10,389 @@
 
 grafo* grafo_novo()
 {
-   return NULL;
+    grafo* g = (grafo*)malloc(sizeof(grafo));
+    
+    if(g == NULL){
+        return NULL;
+    }
+
+    g->nos = (no_grafo**) malloc(sizeof(no_grafo));
+    
+    if(g->nos == NULL){
+        free(g);
+        return NULL;
+    }
+
+    g->tamanho = 0;
+
+    return g;
 }
 
 
 void grafo_apaga(grafo* g)
 {
-  
+    if(g == NULL){
+        return;
+    }
+
+    int j = 0;
+
+    if(g->tamanho > 0){
+        for (int i=0;i<g->tamanho;i++)
+        {
+            while(g->nos[i]->tamanho > j)

```

Comment:
This second loop is worrisome, my expectation is that the full array of nodes is contained in the grafo structure.

The only thing to deallocate in the node should be the connection list.